mmv1/api/type.rb

```diff
@@ -637,6 +637,7 @@ class ResourceRef < Type
       # The fields which can be overridden in provider.yaml.
       module Fields
         attr_reader :resource
+        attr_reader :resourceproduct
```

Comment:
Am I missing something with this field? I don't think we ever set it (and it's redundant with @__resource.__product anyways)

Reply:
> Additionally, the type.rb file has been updated to have an output property that directly contains the suppled "resource", "resourceproduct" and "imports" strings from the yaml attributes. Nothing uses them at this time, but it makes those pieces more available to potential custom_* functions in the future.

The idea was it could be set if the resource reference was to resource in a seperate product, and thus make it accessible downstream through the supplied_values. I had it at first added while trying to preserve *some* of the validation functionality, but kept it after removing more as it felt like something that could be useful. Could just as well remove it and the remaining validation that checks if a resource exists within current product should that be the case, as it was trying to cover the edge case of "two resources in separate products have the same name" without requiring other custom code.

Reply:
Sorry, failure to read on my part! Read the first two paragraphs and then got distracted by the failing builds.

> Could just as well remove it and the remaining validation that checks if a resource exists within current product should that be the case, as it was also trying to cover the one edge case of "two resources in separate products have the same name" without requiring other custom code.

I'd lean towards removing it. Ambiguous references like that are probably going to be generic, in which case they'll be full resource name (https://google.aip.dev/122#full-resource-names) anyways, so that's not something we need to solve for in the average case. If we want the functionality in the future, we can just add it in at that point!